kubernetes/FAQs.md

```diff
@@ -0,0 +1,84 @@
+## Introduction
+
+From Kubernetes v1.13, CoreDNS is the recommended default DNS server in Kubernetes. 
+
+As more and more users are deploying and using CoreDNS is their Kubernetes cluster, there are a number of FAQs asked by users having trouble setting up CoreDNS, or maintaining CoreDNS. 
+This document aims to maintain a collection of the most frequently asked questions by users during the support provided to users having trouble with the DNS Service after installation or upgrade of their Kubernetes cluster.
+
+## Useful links about CoreDNS for Kubernetes:
+
+- [Using CoreDNS for Service Discovery](https://kubernetes.io/docs/tasks/administer-cluster/coredns/)
+- [Debugging DNS Resolution](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/)
+- [Troubleshooting kubeadm](https://kubernetes.io/docs/setup/independent/troubleshooting-kubeadm/)
+- [DNS Specifications for Kubernetes](https://github.com/kubernetes/dns/blob/master/docs/specification.md)
+- [CoreDNS Kubernetes plugin documentation](https://coredns.io/plugins/kubernetes/)
+
+## Pre-requisites before troubleshooting
+
+It is important to check whether the Kubernetes Network is stable before starting to troubleshoot DNS. It may appear like DNS isn't functioning when there is a network issue in the Kubernetes cluster. 
+In other words, an issue like "DNS does not work properly" has often its root cause as "Network is not working properly".
+
+Hence, if you're confident that the network is stable and DNS still doesn't work, this FAQ document is applicable. 
+
+## Known Issues
+
+### Some Prometheus data is missing after reload
+
+In CoreDNS v1.2.2 that is provided with K8s v1.12, after editing the ConfigMap,
+CoreDNS will automatically "reload" its configuration file.
+
+The reload causes some metrics on the Prometheus interface to be missing.
+
+## Frequently Asked Questions
+
+#### FAQ-1 : Why am I unable to get an answer to the DNS queries when I'm using ```busybox``` to test?
+
+Busybox has a [known bug](https://github.com/docker-library/busybox/issues/48) and therefore does not work for testing DNS queries in Kubernetes.
+The workaround for this is to ensure that the image of Busybox is lower or equal to 1.28.4
+See: [dns can't resolve kubernetes.default and/or cluster.local](https://github.com/kubernetes/kubernetes/issues/66924#issuecomment-411804435) for more details.
+
+#### FAQ-2 : Why isn't the `autopath` plugin in the default configuration?
+
+Autopath requires  that the option `pods verified` is set in Kubernetes's plugin configuration. The `pods verified` takes up significantly more memory. Please look into the [documentation](https://github.com/coredns/coredns/tree/master/plugin/kubernetes#autopath) for more details.
+
+#### FAQ-3 : How do I test the DNS service?
+
+There are two definite ways to test this: 
```

Comment:
Perhaps mention that the important thing is that you are testing dns resolution from a Pod, not from one of the nodes.  Two examples of doing this are to use a dnstools pod, or to use the busybox pod.